syft/workers/base.py

```diff
@@ -401,7 +402,13 @@ def send(
 
         return pointer
 
-    def execute_command(self, message: tuple) -> PointerTensor:
+    def execute_command(self, message) -> PointerTensor:
+        if isinstance(message, ComputationAction):
+            return self.execute_computation(message)
+        else:
+            return self.execute_communication(message)
+
+    def execute_computation(self, message) -> PointerTensor:
```

Comment:
```suggestion
    def execute_computation(self, action: ComputationAction) -> PointerTensor:
```
If `Action` is the type, `message` might not be the right name for these parameters.

Reply:
Same